vignettes/endpoints.Rmd

```diff
@@ -0,0 +1,392 @@
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/nidss_dengue.html>
+
+```{r}
+nidss_dengue(locations = "taipei", epiweeks = epirange(200301, 201301)) %>% fetch_tbl()
+```
+
+## PAHO Dengue
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/paho_dengue.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/paho_dengue.html>
+
+```{r, eval=FALSE}
+paho_dengue(regions = "ca", epiweeks = epirange(200201, 202319)) %>% fetch_tbl()
+```
+
+## Other Endpoints
+
+### Wikipedia Access
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/wiki.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/wiki.html>
+
+```{r}
+wiki(language = "en", articles = "influenza", epiweeks = epirange(202001, 202319)) %>% fetch_tbl()
+```
+
+# Private methods
+
+These require private access keys to use. To actually run these locally, you will need to store these secrets in your `.Reviron` file, or set them as environmental variables.
+
+### CDC
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/cdc.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/pvt_cdc.html>
+
+Example call:
+
+```{r, eval=FALSE}
+pvt_cdc(auth = Sys.getenv("SECRET_API_AUTH_CDC"), epiweeks = epirange(202003, 202304), locations = "ma") %>% fetch_tbl()
+```
+
+### Dengue Digital Surveillance Sensors
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/dengue_sensors.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/dengue_sensors.html>
+
+Example call:
+
+```{r, eval=FALSE}
+pvt_dengue_sensors(auth = Sys.getenv("SECRET_API_AUTH_SENSORS"), names = "ght", locations = "ag", epiweeks = epirange(201404, 202004)) %>% fetch_tbl()
+```
+
+### Google Health Trends
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/ght.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/pvt_ght.html>
+
+Example call:
+
+```{r, eval=FALSE}
+pvt_ght(auth = Sys.getenv("SECRET_API_AUTH_GHT"), epiweeks = epirange(199301, 202304), locations = "ma", query = "how to get over the flu") %>% fetch_tbl()
+```
+
+### NoroSTAT metadata
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/meta_norostat.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/pvt_meta_norostat.html>
+
+Example call:
+
+```{r, eval=FALSE}
+pvt_meta_norostat(auth = Sys.getenv("SECRET_API_AUTH_NOROSTAT")) %>% fetch_classic()
+```
+
+### NoroSTAT data
+
+API docs: <https://cmu-delphi.github.io/delphi-epidata/api/norostat.html>
+
+Function reference: <https://cmu-delphi.github.io/epidatr/reference/pvt_norostat.html>
+
+Example call:
+
+```{r, eval=FALSE}
+pvt_norostat(auth = Sys.getenv("SECRET_API_AUTH_NOROSTAT"), locations = "ma", epiweeks = epirange(199001, 202105)) %>% fetch_classic()
+```
+
+TODO can't seem to get a relevant sample, the example in the reference section is borked.
```

Comment:
Were you able to find a working example for NoroSTAT?

Reply:
I see your comment about the database now. I looked at the server code and the db is `norostat_point_diffs`. Could you look at that one?

Reply:
thought I'd checked that one, but going through it again, it still gives a `-2`; actually, the API directly via `https://api.delphi.cmu.edu/epidata/norostat/?auth=yourAuth&epiweeks=201233&location=1
` gives a -2, even though I get
```mysql
mysql> SELECT * FROM norostat_point_diffs LIMIT 10;
+--------------+---------------------+-------------+---------+-----------+
| release_date | parse_time          | location_id | epiweek | new_value |
+--------------+---------------------+-------------+---------+-----------+
| 2014-10-21   | 2018-10-08 15:24:51 |           1 |  201231 | 4         |
| 2015-12-21   | 2018-10-08 15:24:53 |           1 |  201231 | NULL      |
| 2014-10-21   | 2018-10-08 15:24:51 |           1 |  201232 | 7         |
| 2015-12-21   | 2018-10-08 15:24:53 |           1 |  201232 | NULL      |
| 2014-10-21   | 2018-10-08 15:24:51 |           1 |  201233 | 2         |
| 2015-12-21   | 2018-10-08 15:24:53 |           1 |  201233 | NULL      |
| 2014-10-21   | 2018-10-08 15:24:51 |           1 |  201234 | 5         |
| 2015-12-21   | 2018-10-08 15:24:53 |           1 |  201234 | NULL      |
| 2014-10-21   | 2018-10-08 15:24:51 |           1 |  201235 | 3         |
| 2015-12-21   | 2018-10-08 15:24:53 |           1 |  201235 | NULL      |
+--------------+---------------------+-------------+---------+-----------+
```

So this may be a problem upstream

Reply:
Alright, well for the time being let's put norostat request arguments that _should_ work in this vignette and we can recompile it later when upstream problems are resolved.